syntax = "proto3";

package project;

// Interface exported by the server.

service SVM {
    rpc GetWeights(Row) returns (Null) {}
    rpc SendNodeInfo(Node_Config) returns (Null) {}
    rpc SendLearningInfo(Learning_Config) returns (Null) {}
    rpc SendCompletionSignal(Null) returns (Null) {}
}

message Null { }

message Node_Config {
    string coordinator_address = 1;
    repeated string workers_address = 2;
}

message Learning_Config {
    float lr = 1;
    float tot_iter = 2;
    int32 batch_size = 3;
    float reg = 4;
}

message Row {
    string label = 1;
    repeated Entry entries = 2;
}

//message WeightUpdate{
//    Row row = 1;
//}

message Status {
    string status = 1;
}

message Entry{
    int32 index = 1;
    double value = 2;
}
